extensions/notebook/src/dialog/managePackages/installedPackagesTab.ts

```diff
@@ -112,6 +111,51 @@ export class InstalledPackagesTab {
 		});
 	}
 
+	public async resetLocations(): Promise<void> {
+		if (this.view) {
+			if (this.locationComponent) {
+				this.formBuilder.removeFormItem({
+					component: this.locationComponent,
+					title: localize('managePackages.location', "Location")
+				});
+			}
+			const locations = await this.dialog.model.getLocations();
+			if (locations && locations.length === 1) {
+				this.locationComponent = this.view.modelBuilder.text().withProperties({
+					value: locations[0].displayName
+				}).component();
+			} else if (locations) {
+				let dropdownValues = locations.map(x => {
+					return {
+						name: x.name,
+						displayName: x.displayName
+					};
+				});
+				let locationDropDown = this.view.modelBuilder.dropDown().withProperties({
+					values: dropdownValues,
+					value: dropdownValues[0]
+				}).component();
+				this.dialog.changeLocation(dropdownValues[0].name);
+				locationDropDown.onValueChanged(() => {
+					this.dialog.changeLocation((<azdata.CategoryValue>locationDropDown.value).name);
+					this.dialog.resetPages()
+						.catch(err => {
+							this.dialog.showErrorMessage(utils.getErrorMessage(err));
+						});
+				});
+				this.locationComponent = locationDropDown;
+			} else {
+				this.locationComponent = this.view.modelBuilder.text().withProperties({
```

Comment:
Should we display some kind of error message in this case?

Reply:
I'm trying to support not specifying the location. I have to change the APIs for that too but that's why I didn't want to throw error here

---
In reply to: [400516061](https://github.com/microsoft/azuredatastudio/pull/9790#discussion_r400516061) [](ancestors = 400516061)